modules/custom/az_migration/src/Plugin/migrate/source/AZNodeWithFieldCollection.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\source;
+
+use Drupal\migrate\Row;
+use Drupal\node\Plugin\migrate\source\d7\Node as D7Node;
+
+/**
+ * Extends D7Node source plugin to convert field collection content into paragraph content.
```

Comment:
Nitpick:
```suggestion
 * Node source plugin that adds field collection data for the given field names.
```

Reply:
Agreed

Reply:
Should we say not to use this to migrate nodes, just field collections on nodes?

Reply:
Since it's a source plugin, I don't think we should specify what destinations it is used for.

Reply:
How about 'Node source plugin that migrates field collection data into paragraphs' ... or something that to that effect. When I read these source plugin descriptions I really want to know what it's trying to accomplish for me.

Reply:
Maybe you can get that from the rest of the comments / context, hard to say.